### PR TITLE
web: fix gitlab range link

### PR DESCRIPTION
**Pull request by anukul**
Fixes https://github.com/sourcegraph/sourcegraph/issues/12718

### CODE REVIEW

**Comment by sqs** [comment]
@anukul Thank you!

@sourcegraph/web Could you please review this? It looks like a nice fix.

File: web/src/repo/actions/GoToCodeHostAction.tsx [hunk 1]
```diff
@@ -127,7 +127,8 @@ export class GoToCodeHostAction extends React.PureComponent<Props, State> {
             }
             // Add range or position path to the code host URL.
             if (this.props.range) {
-                url += `#L${this.props.range.start.line}-L${this.props.range.end.line}`
+                const rangeEndPrefix = externalURL.serviceType === 'github' ? 'L' : ''
```
Comment (felixfbecker) [inline web/src/repo/actions/GoToCodeHostAction.tsx:130]: This doesn't just remove the `L` for GitLab, but also from Bitbucket, Phabricator, etc. Do those URLs also not use the `L` prefix?

Comment (felixfbecker) [reply web/src/repo/actions/GoToCodeHostAction.tsx:130]: If you are unsure, I'd prefer to change it _only_ for Gitlab and keep the behavior the same for all other code hosts.

Comment (anukul) [reply web/src/repo/actions/GoToCodeHostAction.tsx]: @felixfbecker It looks like Bitbucket has another different [syntax](https://stackoverflow.com/a/59745510/4077900) for this. I noticed that the line number suffix is only added to GitHub and GitLab URLs at the moment, but I have made the change as suggested.

**Pull request merged**